vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,404 @@ title: Registries concepts
+
+For each port in the registry, a corresponding _version file_ must exist.
+The versions file is a JSON file with the same name as its corresponding port,
+for example, a port named `foo` must have a corresponding `foo.json` version
+file.
+
+vcpkg expects the versions files to be stored at the following location
+`versions/<prefix>/<port name>.json` inside the registry, where `<prefix>` is
+the first letter of the port name followed by a hyphen. For example, the
+version file for port `foo` must be stored in `versions/f-/foo.json`.
+
+The purpose of the version file is two-fold:
+
+* List all available versions of each port
+* Point to the retrieval locations of each of version.
+
+Same as the baseline file, the layout of the version file depends on the kind of
+registry.
+
+## Baseline file
 
-There are currently two options to implement your own registries: a Git-based
-registry or a filesystem-based registry.
+All registries, regardless of their kind, must contain a file named
+`baseline.json` located at `versions/baseline.json`.
+
+The purpose of the baseline file is to describe the set of versions that are
+considered to be the latest for all ports in the registry. It is expected that
+this file is updated each time a new version of a port is added to the registry.
+
+The layout of the file is a map of named baselines. With each named baseline
+being itself a map of port names to version entries.
+
+Each baseline version entry is an object with the following properties:
+
+* `baseline`: the value is the port's version matching its corresponding
+  `vcpkg.json` file.
+* `port-version`: the value is the port's `port-version` matching its
+  corresponding `vcpkg.json` file.
+
+### Example baseline file
+
+```json
+{
+  "default": {
+    "foo": { "baseline": "1.0.0", "port-version": 0 },
+    "bar": { "baseline": "2024-08-01", "port-version": 1 },
+    "baz": { "baseline": "vista-xp", "port-version": 0 },
+  }
+}
+```
+
+See the reference documentation for:
+
+* baseline file layout for [Git
+  registries](../maintainers/registries.md#git-baseline)
+* baseline file layout for [filesystem
+  registries](../maintainers/registries.md#filesystem-baseline)
+
+## Version files
+
+Each port in the registry must have a corresponding versions file. The versions
+file is a JSON file named the same as its corresponding port, for example, a
+port named `foo` must have a corresponding `foo.json` file. 
+
+vcpkg expects the versions files to be stored in the following location
+`versions/<prefix>/<port name>.json` , where `<prefix>` is the first letter of
+the port name followed by a hyphen. For example, the versions file for port
+`foo` must be stored in `versions/f-/foo.json`.
+
+The purpose of the versions file is two-fold:
+
+* List all available versions of each port
+* Point to the retrieval locations of each of these versions.
+
+The layout of the version file is an object containing a `"versions"` array, with
```

Comment:
Should we write a JSON-schema for these things? (No change requested in this PR, just wondering)